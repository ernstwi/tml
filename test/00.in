init foo
foo -> bar
    guard { x - y == 5 }
bar { x < 10 }
bar -> baz
