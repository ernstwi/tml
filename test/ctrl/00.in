init a

a >> b
b -> c
    action { foo }
