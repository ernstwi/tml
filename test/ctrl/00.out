{
    "actions": [
        "foo"
    ],
    "clocks": [],
    "edges": [
        {
            "controllable": false,
            "from": "a",
            "guard": null,
            "reset": null,
            "sync": null,
            "to": "b"
        },
        {
            "controllable": true,
            "from": "b",
            "guard": null,
            "reset": null,
            "sync": {
                "id": "foo",
                "type": "internal"
            },
            "to": "c"
        }
    ],
    "locations": [
        {
            "id": "a",
            "initial": true,
            "invariant": null
        },
        {
            "id": "b",
            "initial": false,
            "invariant": null
        },
        {
            "id": "c",
            "initial": false,
            "invariant": null
        }
    ]
}
